src/MultiFormats/HexString.cs

```diff
@@ -0,0 +1,161 @@
+                    }
+                })
+               .Distinct()
+               .ToDictionary(v => v.String, v => (byte) v.Value);
+
+        /// <summary>
+        ///   Converts an array of 8-bit unsigned integers to its equivalent hexadecimal string representation.
+        /// </summary>
+        /// <param name="buffer">
+        ///   An array of <see cref="byte">8-bit unsigned integers</see>.
+        /// </param>
+        /// <param name="format">
+        ///   One of the format specifiers ("G" and "x" for lower-case hex digits, or "X" for the upper-case).
+        ///   The default is "G".
+        /// </param>
+        /// <returns>
+        ///   The string representation, in hexadecimal, of the contents of <paramref name="buffer"/>.  
+        /// </returns>
+        public static string Encode(byte[] buffer, string format = "G")
+        {
+            string[] hexStrings;
+            switch (format)
+            {
+                case "G":
+                case "x":
+                    hexStrings = LowerCaseHexStrings;
+                    break;
+                case "X":
+                    hexStrings = UpperCaseHexStrings;
+                    break;
+                default:
+                    throw new FormatException(
+                        string.Format("Invalid HexString format '{0}', only 'G', 'x' or 'X' are allowed.", format));
+            }
+
+            var s = new StringBuilder(buffer.Length * 2);
+            foreach (var v in buffer)
+                s.Append(hexStrings[v]);
+            return s.ToString();
+        }
+
+        /// <summary>
+        ///   Converts an array of 8-bit unsigned integers to its equivalent hexadecimal string representation.
+        /// </summary>
+        /// <param name="buffer">
+        ///   An array of <see cref="byte">8-bit unsigned integers</see>.
+        /// </param>
+        /// <param name="format">
+        ///   One of the format specifiers ("G" and "x" for lower-case hex digits, or "X" for the upper-case).
+        ///   The default is "G".
+        /// </param>
+        /// <returns>
+        ///   The string representation, in hexadecimal, of the contents of <paramref name="buffer"/>.
+        /// </returns>
+        public static string ToHexString(this byte[] buffer, string format = "G") { return Encode(buffer, format); }
+
+        /// <summary>
+        ///   Converts the specified <see cref="string"/>, which encodes binary data as hexadecimal digits, 
+        ///   to an equivalent 8-bit unsigned integer array.
+        /// </summary>
+        /// <param name="s">
+        ///   The hexadecimal string to convert.
+        /// </param>
+        /// <returns>
+        ///   An array of 8-bit unsigned integers that is equivalent to <paramref name="s"/>.
+        /// </returns>
+        public static byte[] Decode(string s)
+        {
+            var n = s.Length;
+            if (n % 2 != 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979568)